src/lib/operators/sort.cpp

```diff
@@ -33,19 +20,77 @@ const std::string& Sort::name() const {
 std::shared_ptr<AbstractOperator> Sort::_on_deep_copy(
     const std::shared_ptr<AbstractOperator>& copied_input_left,
     const std::shared_ptr<AbstractOperator>& copied_input_right) const {
-  return std::make_shared<Sort>(copied_input_left, _column_id, _order_by_mode, _output_chunk_size);
+  return std::make_shared<Sort>(copied_input_left, _sort_definitions, _output_chunk_size);
 }
 
 void Sort::_on_set_parameters(const std::unordered_map<ParameterID, AllTypeVariant>& parameters) {}
 
 std::shared_ptr<const Table> Sort::_on_execute() {
-  _impl = make_unique_by_data_type<AbstractReadOnlyOperatorImpl, SortImpl>(
-      input_table_left()->column_data_type(_column_id), input_table_left(), _column_id, _order_by_mode,
-      _output_chunk_size);
-  return _impl->_on_execute();
+  _sort_definition_data_types.resize(_sort_definitions.size());
+  std::transform(
+      _sort_definitions.begin(), _sort_definitions.end(), _sort_definition_data_types.begin(),
+      [this](const auto sort_definition) { return input_table_left()->column_data_type(sort_definition.column); });
+
+  std::shared_ptr<const Table> output = input_table_left();
+  std::shared_ptr<Table> output_mutable;
+  std::shared_ptr<PosList> previously_sorted_pos_list = std::shared_ptr<PosList>(nullptr);
+
+  _validate_sort_definitions();
+
+  for (auto column_id = _sort_definitions.size(); column_id-- != 0;) {
+    const bool is_last_sorting_run = (column_id == 0);
+
+    auto sort_definition = _sort_definitions[column_id];
+    DataType data_type = _sort_definition_data_types[column_id];
+
+    resolve_data_type(data_type, [&](auto type) {
+      using ColumnDataType = typename decltype(type)::type;
+
+      auto sort_single_col_impl = SortImpl<ColumnDataType>(input_table_left(), sort_definition.column,
```

Comment:
s/col/column/

Reply:
Hm not sure what you're aiming at, is ist about the naming? So sort_s_column_impl would be better from your point of view?

Reply:
My fault. Shouldn't use [sed syntax](https://unix.stackexchange.com/questions/159367/using-sed-to-find-and-replace) in comments. "Replace col with column" would have been not that much worse to type. ;)